ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -687,7 +742,7 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
           , topLevelConfigBlock    = blockConfigByron
           }
       , pInfoInitLedger = initExtLedgerStateByron
-      } = protocolInfoByron protocolParamsByron
+      } = protocolInfoByron paramsByron
```

Comment:
INFO: the `protocolInfo*` functions can probably abstracted over too, into something like a `MakeProtocolInfo` class. Cardano tools already provide some form of abstraction for this, so it probably makes sense to do it here too, and then cardano tools can reuse the machinery. If reviewers agree, I'll create a separate issue for this.

Reply:
It's not obvious to me that we'd benefit from abstracting over the way to create protocol infos for different block types. Their differences seem too "ad-hoc".

Reply:
There are also some things that look less composable at a glance, like how `initExtLedgerStateCardano` is created, and how Byron `BlockForging` is merged with Shelley `BlockForging`.

Reply:
Maybe we can't define a `protocolInfoHardForkBlock` for the general `HardForkBlock`, but we could try refactoring `protocolInfoCardano` a bit so it is more in the style of "make a protocol info for each individual block, and then do the ad-hoc composing"

Reply:
I think your last comment would be a good first step.